src/app/zap-templates/endpoint_config.zapt

```diff
@@ -3,4 +3,100 @@
+#define ZAP_ATTRIBUTE_INDEX(index) ((EmberAfAttributeMetadata *)(&generatedAttributes[index]))
+
+#define GENERATED_CLUSTER_COUNT {{endpoint_cluster_count}}
+#define GENERATED_CLUSTERS {{endpoint_cluster_list}}
+
+#define ZAP_CLUSTER_INDEX(index) ((EmberAfCluster*)(&generatedClusters[index]))
+
+#define ZAP_CLUSTER_MASK(mask) CLUSTER_MASK_ ## mask
+// This is an array of EmberAfEndpointType structures.
+#define GENERATED_ENDPOINT_TYPES {{endpoint_types_list}}
+
+
+// Largest attribute size is needed for various buffers
+#define ATTRIBUTE_LARGEST ({{endpoint_largest_attribute_size}})
+
+// Total size of singleton attributes
+#define ATTRIBUTE_SINGLETONS_SIZE ({{endpoint_singletons_size}})
+
+// Total size of attribute storage
+#define ATTRIBUTE_MAX_SIZE ({{endpoint_total_storage_size}})
+
+// Number of fixed endpoints
+#define FIXED_ENDPOINT_COUNT ({{endpoint_count}})
+
+// Array of endpoints that are supported, the data inside
+// the array is the endpoint number.
+#define FIXED_ENDPOINT_ARRAY {{endpoint_fixed_endpoint_array}}
+
+// Array of profile ids
+#define FIXED_PROFILE_IDS {{endpoint_fixed_profile_id_array}}
+
+// Array of device ids
+#define FIXED_DEVICE_IDS {{endpoint_fixed_device_id_array}}
+
+// Array of device versions
+#define FIXED_DEVICE_VERSIONS {{endpoint_fixed_device_version_array}}
+
+// Array of endpoint types supported on each endpoint
+#define FIXED_ENDPOINT_TYPES {{endpoint_fixed_endpoint_type_array}}
+
+// Array of networks supported on each endpoint
+#define FIXED_NETWORKS {{endpoint_fixed_network_array}}
+
+
+// Array of EmberAfCommandMetadata structs.
+#define ZAP_COMMAND_MASK(mask) COMMAND_MASK_ ## mask
+#define EMBER_AF_GENERATED_COMMAND_COUNT ({{endpoint_command_count}})
+#define GENERATED_COMMANDS {{endpoint_command_list}}
+
+// Array of EmberAfManufacturerCodeEntry structures for commands.
+#define GENERATED_COMMAND_MANUFACTURER_CODE_COUNT ({{endpoint_command_manufacturer_code_count}})
+#define GENERATED_COMMAND_MANUFACTURER_CODES {{endpoint_command_manufacturer_codes}}
+
+// This is an array of EmberAfManufacturerCodeEntry structures for clusters.
+#define GENERATED_CLUSTER_MANUFACTURER_CODE_COUNT ({{endpoint_cluster_manufacturer_code_count}})
+#define GENERATED_CLUSTER_MANUFACTURER_CODES {{endpoint_cluster_manufacturer_codes}}
+
+// This is an array of EmberAfManufacturerCodeEntry structures for attributes.
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODE_COUNT ({{endpoint_attribute_manufacturer_code_count}})
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODES {{endpoint_attribute_manufacturer_codes}}
+
+// Array of EmberAfPluginReportingEntry structures.
+#define ZRD(x) EMBER_ZCL_REPORTING_DIRECTION_ ## x
+#define ZAP_REPORT_DIRECTION(x) ZRD(x)
+
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS_TABLE_SIZE ({{endpoint_reporting_config_default_count}})
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS {{endpoint_reporting_config_defaults}}
+
+
+{{/endpoint_config}}
```

Comment:
All the `#define EMBER_AF_GENERATED_PLUGIN_*` code seems to be gone.

I understand the issue to generate those, but we need to find a solution, we can not just remove them.

Reply:
@vivien-apple Could EMBER_AF_GENERATED_PLUGIN_* be replace with a generated cluster define instead ? Something like CHIP_GENERATED_CLUSTER_*cluster_name* ? Or is there a use case where this solution wouldn't work ?

Reply:
@vivien-apple a temporary solution was implemented. We should discuss about a more permanent one after the Holidays in a software TT meeting.

Reply:
Any reasons to not uses the macros from `gen_config.h` to check for the plugins in `util.cpp` instead of adding new macros in `endpoint_config.h` ?

Reply:
A more specific example. Can you use `EMBER_AF_PLUGIN_TEMPERATURE_MEASUREMENT_SERVER` instead of `CHIP_TEMP_MEASUREMENT_CLUSTER_SERVER` ?